ztrouy/frontend-capstone-api
# Feature - Initial Group Views

### Purpose
To allow Users to see, create, edit, and delete groups

### Files Changed
- Added method that returns the list of Games that all Users in a Group share that can also support as many Users as exist in the Group in `group.py`
- Created file, wrote serializers, and wrote initial CRUD views for Groups in new `groups.py`
- Imported Groups View Set in the `__init__.py` package in the `/views` directory
- Added routing for the Groups View Set in `urls.py`

### To Test
Fetch, setup by running `pipenv shell` and `./seed_database.sh` in the project root directory, run the debugger, then confirm the following
- GET endpoint `/groups` returns a list of all Groups, with each having an Id, Name, Member Count, and boolean specifying if the User who made the request is a Member of the Group or not
- GET endpoint `/groups/{id}` returns the appropriate Group if it exists, including all of the information above, a list of all Members, and the list of shared Games all Members own that support the number of Members in the Group
- POST endpoint `/groups` creates a new Group when provided with valid JSON, adds the User that made it to the Group, and then returns the Group
- PUT endpoint `/groups/{id}` that updates the appropriate Group if it exists and is provided with valid JSON
- DELETE endpoint `/groups/{id}` that deletes the appropriate Group if it exists

closes #9
